presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -684,32 +693,40 @@ public void end()
         void afterValue();
     }
 
+    private interface BlockFieldConverter extends BlockConverter
```

Comment:
Is it possible to update all BlockConverter into BlockFieldConverter? Then we just need on BlockConverter, with fieldIndex